### PR TITLE
Yarn workspaces with lerna

### PR DESCRIPTION
**Pull request by echarles**
This PR will implemen #15 for Yarn workspaces and Lerna.

This is still WIP but current version can already be used

```
yarn bootstrap
yarn build-ext
yarn link-ext
yarn build-jlab
yarn clean-ext
```

As discussed, separated extensions can still be used and developed as before.

### CODE REVIEW

**Comment by echarles** [comment]
This PR also gives a namespace to the packages: `hello_world` -> `@jupyterlab-examples/hello_world`

I propose to wait before merging this (after relevant reviews) the other PRs and take in this one the move of the current folder under a `packages` folder to avoid the exhaustive listing of the folders in the `package.json` and `lerna.json`.

Better maybe, we could already introduce some extension families like `base/*`, `advanced/
*`, `react/*`or by [extension points](https://jupyterlab.readthedocs.io/en/stable/developer/extension_points.html). In that case we would define in `package.json`

```json
"packages": {
"commands/*",
"command_palette/*",
"main_menu/*",
"context_menu/*",
"keyboard_shortcuts/*",
"launcher/*",
"lr_areas/*",
"status_bar/*",
"widget_tracker/*",
"copy_shareable_link/*"
"basics/*",
"advances/*",
"react/*",
}

@jtpio WDYT?

**Comment by jtpio** [comment]
Thanks @echarles, I like the idea :+1: 

> This PR also gives a namespace to the packages: `hello_world` -> `@jupyterlab-examples/hello_world`

Looks good. Should we also replace the `_` by `-`? So `hello_world` becomes `@jupyterlab-examples/hello-world`

> Better maybe, we could already introduce some extension families like `base/*`, `advanced/ *`, `react/*`or by [extension points](https://jupyterlab.readthedocs.io/en/stable/developer/extension_points.html). In that case we would define in `package.json`

It would be really nice to have this connection to the existing extension points. Note sure yet about the difficulty, as it might sometimes be subjective?

We could also rebase this PR so it picks up the [CI job](https://github.com/jtpio/jupyterlab-extension-examples/blob/master/.github/workflows/main.yml) (added in #11), and see if the workflow works on CI too?

**Comment by echarles** [comment]
@jtpio 

> Should we also replace the _ by -? So hello_world becomes @jupyterlab-examples/hello-world

I was thinking the same. Now it is done.

> It would be really nice to have this connection to the existing extension points. Note sure yet about the difficulty, as it might sometimes be subjective?

I have pushed a structure sekeleton and dispatched the examples into it. It may need additional passes, especially having finer grained and also more meta examples. This can be done in other PRs. The important point with this PR is to have a skeleton structure with yarn workspaces/lerna we can work on.

I have also updated the README to show how to work with all examples or work with one by one. 

> We could also rebase this PR so it picks up the CI job (added in #11), and see if the workflow works on CI too?

I think it was already done, I see now that CI is failing. Let me fix the config and try again.

**Comment by echarles** [comment]
@jtpio CI is updated and now builds also JupyterLab. All green.

LGTM

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -33,13 +32,20 @@ jobs:
...
+    - name: Install the Python dependencies
       run: python -m pip install jupyterlab
-    - name: Install the extension
+    - name: Bootstrap the jlpm deps
+      run: jlpm
+    - name: Build all the extensions
+      run: |
+        jlpm build-ext
+        jlpm link-ext
+        jlpm build-jlab
```
Comment (jtpio) [inline .github/workflows/main.yml]: We might want to move this to a `build_all` job, but it's fine if we do it in a separate PR.

File: README.md [hunk 2]
```diff
@@ -12,30 +12,37 @@ git clone https://github.com/jtpio/jupyterlab-extension-examples.git &&
...
-+ [Jupyter Widgets](./jupyter_widgets)
-+ [Kernel Output](./kernel_output)
-+ [Kernel Messaging](./kernel_messaging)
-+ [Serving Files](./serving_files)
+Basics
++ [Hello World](./basics/hello-world)
++ [Signals](./basics/signals)
++ [Datagrid](./basics/datagrid)
+
+Contex Menu
```
Comment (jtpio) [inline README.md]:
```suggestion
Context Menu
```

File: README.md [hunk 3]
```diff
@@ -89,15 +112,15 @@ jupyter lab --watch
...
 
-Complementary, you can rely on the official JupyterLab documentation.
+Complementary to these examples, you can rely on the official JupyterLab documentation.
 
-* https://jupyterlab.readthedocs.io/en/stable/developer/extension_dev.html
-* https://jupyterlab.readthedocs.io/en/stable/developer/extension_points.html
-* http://jupyterLab.readthedocs.io/en/stable/developer/extension_tutorial.html
+* [Extension Developer Guide](https://jupyterlab.readthedocs.io/en/stable/developer/extension_dev.html)
+* [Common Extension Points](https://jupyterlab.readthedocs.io/en/stable/developer/extension_points.html)
+* [Astronomy Picture of the Day JupyterLab Extension](http://jupyterLab.readthedocs.io/en/stable/developer/extension_tutorial.html)
```
Comment (jtpio) [inline README.md]:
```suggestion
* [Astronomy Picture of the Day JupyterLab Extension](https://jupyterlab.readthedocs.io/en/stable/developer/extension_tutorial.html)
```

**Comment by jtpio** [comment]
This is great, thanks @echarles!

Left a couple of inline comments and then :rocket:

File: .github/workflows/main.yml (hunk 1 above)
Comment (echarles) [reply .github/workflows/main.yml]: Have created a separated build_all.yml in `.github/workflows` - I am not used to GitHub CI so I will push and see how it behaves

**Comment by echarles** [comment]
@jtpio All green with separated build_all
